packages/grid/src/vaadin-grid-keyboard-navigation-mixin.js

```diff
@@ -108,11 +108,20 @@ export const KeyboardNavigationMixin = (superClass) =>
     }
 
     set __rowFocusMode(value) {
-      ['_itemsFocusable', '_footerFocusable', '_headerFocusable'].forEach((focusable) => {
-        if (value && this.__isCell(this[focusable])) {
-          this[focusable] = this[focusable].parentElement;
-        } else if (!value && this.__isRow(this[focusable])) {
-          this[focusable] = this[focusable].firstElementChild;
+      ['_itemsFocusable', '_footerFocusable', '_headerFocusable'].forEach((prop) => {
+        const focusable = this[prop];
+        if (value) {
+          const parent = focusable && focusable.parentElement;
+          if (this.__isCell(focusable)) {
+            // Cell itself focusable (default)
+            this[prop] = parent;
+          } else if (this.__isCell(parent)) {
+            // Div inside a cell is focusable
```

Comment:
This is a bit generic IMO, we could point out that this explicitly related to the new mode:
```suggestion
            // {whatever mode name is} mode is enabled, button element inside the cell is focusable
```

Reply:
Done.